### DISCUSSION TITLE
support coroutines

### DISCUSSION THREAD

**Discussion by commonuserlol** [vfsfitvnm/frida-il2cpp-bridge#325] [Ideas]
hey, sometimes devs uses coroutines (return type is `System.Collections.IEnumerator` and MoveNext, Dispose, etc "submethods"). its really headache with their work, so I think in `frida-il2cpp-bridge` we miss `Il2Cpp::coroutine` or smth

**Comment by vfsfitvnm** [comment]
That would be a nice addition, but it's likely I will focus on this anytime soon.
If you have a working code and you just want to avoid writing boilerplate code, that would definitely help me so that I don't have to even try to understand the mechanism underneath

**Reply by commonuserlol** [reply]
in source code it's like https://docs.unity3d.com/Manual/Coroutines.html, but in compiled il2cpp situation worse and my code is `while (Coroutine.method("MoveNext").invoke()) { console.log("bruh") }` 🤣
